wiki/DEBUGGING.md

```diff
@@ -0,0 +1,203 @@
+**Note:** If any issues arise or new commands, tools, or features become useful, please update this guide accordingly!
+
+## Debug.Trace
+
+This library is a quick and effective way to print values of objects in your Haskell Program.
+We will look at `trace` and `traceM`.
+If you want to explore more with this library, please refer to the [documentation](https://hackage.haskell.org/package/base-4.21.0.0/docs/Debug-Trace.html).
+
+`trace :: String -> a -> a`, takes in two inputs: the message and an expression.
+An effective way to utilize `trace` is with the `let` keyword.
+For example, in `Svg.Parser.dotProduct`:
+
+```haskell
+-- | Computes the dot product of two vectors.
+dotProduct :: Vector -> Vector -> Double
+dotProduct v1 v2 =
+    let result = zipWith (*) v1 v2
+    in trace ("Intermediate Products: " ++ show result) (sum result)
+
+ghci> dotProduct [1, 2, 3] [3, 6, 7]
+Intermediate Products: [3.0, 12.0, 21.0]
+```
+
+`traceM :: Applicative f => String -> f ()` is similar to `trace` except the only input is the message, and it returns within any [Applicative Context](https://learnyouahaskell.github.io/functors-applicative-functors-and-monoids.html#applicative-functors).
+This is quite handy to use in do-notation.
+For example, in `Database.CourseQueries.queryCourse`:
+
+```haskell
+-- | Queries the database for all information about @course@, and returns a JSON
+-- object representing the course.
+queryCourse :: T.Text -> IO Value
+queryCourse str = do
+    courseJSON <- returnCourse str
+    traceM ("The course: " ++ show courseJSON)
+    return $ toJSON courseJSON
+
+ghci> queryCourse "CSC324"
+The course: Just (Course {breadth = Just "The Physical and Mathematical...)
+```
+
+**Warning:** One important consideration is how trace is used.
+Haskell has lazy evaluation, so if Debug.Trace's functions are used in a place that does not need to be executed, the output will not be visible.
+Here is an example in `Svg.Parser.dotProduct` that illustrates such a case:
+
+```haskell
+-- | Computes the dot product of two vectors.
+dotProduct :: Vector -> Vector -> Double
+dotProduct v1 v2 =
+    let result = zipWith (*) v1 v2
+        _ = trace ("Intermediate Products: " ++ show result)
+    in (sum result)
+```
+
+## GHCi Debugger
+
+### Setup
+
+Run `stack ghci` to start GHCi.
+This is the [Haskell tool stack](https://downloads.haskell.org/ghc/latest/docs/users_guide/ghci.html#) and there are more features than just this debugger.
+Start by loading the relevant modules using `:l Module`.
+
+### Useful Commands
+
+| **Category**            | **Command**                  | **Description**                                                         |
+| ----------------------- | ---------------------------- | ----------------------------------------------------------------------- |
+| **Setting Up GHCi**     | `stack ghci`                 | Launch the GHCi environment                                             |
```

Comment:
Don't include `stack ghci`, it's not like the others because it runs in the terminal, not within ghci. Plus you've correctly mentioned it in the intro already